File: src/Api/Model/Languageforge/Lexicon/LexPicture.php
<?php

namespace Api\Model\Languageforge\Lexicon;

use Api\Model\Shared\Mapper\ObjectForEncoding;

class LexPicture extends ObjectForEncoding
{
    public function __construct($fileName = '', $guid = '')
    {
        $this->fileName = $fileName;
        $this->caption = new LexMultiText();
        $this->guid = Guid::makeValid($guid);
    }

    /** @var string */
    public $fileName;

    /** @var LexMultiText */
    public $caption;

    /** @var string */
    public $guid;
}

File: src/netcore-api/SIL.XForge.WebApi.Server/Models/Lexicon/LexPicture.cs
namespace SIL.XForge.WebApi.Server.Models.Lexicon
{
    public class LexPicture
    {
        public LexPicture(string fileName = "", string guid = null)
        {
            FileName = fileName;
            Guid = guid ?? System.Guid.NewGuid().ToString();
        }

        public string FileName { get; set; }
        public LexMultiText Caption { get; protected set; } = new LexMultiText();
        public string Guid { get; protected set; }
    }
}
